wd_comp.c

```diff
@@ -464,65 +463,58 @@ int wd_do_comp_sync2(handle_t h_sess, struct wd_comp_req *req)
 	dbg("do, op_type = %hhu, in =%u, out_len =%u\n",
 	    req->op_type, req->src_len, req->dst_len);
 
-	avail_out = req->dst_len;
+	total_avail_out = req->dst_len;
 	/* strm_req and req share the same src and dst buffer */
 	memcpy(&strm_req, req, sizeof(struct wd_comp_req));
 	req->dst_len = 0;
 
 	strm_req.last = 0;
-	while (1) {
-		if (req->src_len > chunk) {
-			strm_req.src_len = chunk;
-			req->src_len -= chunk;
-		} else {
-			strm_req.src_len = req->src_len;
-			req->src_len = 0;
+	do {
+		strm_req.src_len = total_avail_in > chunk ? chunk :
+				   total_avail_in;
+		strm_req.dst_len = total_avail_out > chunk ? chunk :
+				   total_avail_out;
+		if (req->op_type == WD_DIR_COMPRESS) {
+			/* find the last chunk to compress */
+			if (strm_req.src_len < chunk)
+				strm_req.last = 1;
```

Comment:
It seems can remove this.

Reply:
> I'm sorry for my previous comments. The append_store_block() is really unreachable. The only question left for me is how to trigger to a bad status 0x41. I agree the opinion you list about the old logic, but I don't understand how it trigger the error. Can you show me the test case? Maybe I need to add it for CI.

It could be reproduced on my branch (hzhuang1/zip_v0.5.7). I'll try to reproduce it on master branch.